airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -188,6 +197,16 @@ public static ServerRunnable getServer(final ServerFactory apiFactory, final Con
         configs.getWorkspaceRoot(),
         configs.getAirbyteVersionOrWarning(),
         featureFlags);
+    final JobNotifier jobNotifier = new JobNotifier(
```

Comment:
we can shift this down to only be created inside the `featureFlags.usesNewScheduler` block.